dapp/src/components/page/governance/ProposalForm.tsx

```diff
@@ -112,13 +117,51 @@ const ProposalForm: React.FC = () => {
     },
   };
 
+  const isDescriptionValid = (description: string, words: number = 3) => {
+    return description.trim().split(/\s+/).length >= words;
+  };
+
   const submitProposal = async () => {
+    if (
+      !proposalName ||
+      proposalName.length < 10 ||
+      proposalName.length > 256
+    ) {
+      alert("Proposal name is required");
+      return;
+    }
+
+    if (votingDays < 5 || votingDays > 30) {
+      alert("Voting days must be between 5 and 30");
+      return;
+    }
+
+    if (!isDescriptionValid(mdText, 10)) {
+      alert("Proposal description must contain at least 10 words.");
+      return;
+    }
+
+    if (!isDescriptionValid(approveDescription)) {
+      alert("Approved description must contain at least 3 words.");
+      return;
+    }
+
+    if (!isDescriptionValid(rejectDescription)) {
+      alert("Rejected description must contain at least 3 words.");
+      return;
+    }
+
+    if (!isDescriptionValid(cancelledDescription)) {
+      alert("Cancelled description must contain at least 3 words.");
+      return;
+    }
```

Comment:
Actually it's more complicated. Only for approval it's a must to have a description. For reject and cancelled, description is only required if there is a XDR.

While we are here, if no XDR is provided or no description, then in the UI we don't need to show this specific outcome. I expect that a lot of proposals will just have an approve outcome and the rest will be empty. So we don't need to show empty outcomes.